mod-users-bl/src/main/resources/domain/mod-users-bl/features/users-bl.feature

```diff
@@ -77,3 +74,89 @@ Feature: Test user business logic
     When method POST
     Then status 201
     And match responseHeaders contains { 'x-okapi-token': '#present' }
+
+  Scenario: Return a composite object for the currently logged in user
+
+    Given path 'bl-users/_self'
+    When method GET
+    Then status 200
+    And match response.user.active == true
+    And match response.user.personal.firstName == 'Admin'
+    And match response.user.username == 'test-admin'
```

Comment:
This test seems to only check that the API responds and user is the one expected (which also confirms that the tests run as the admin user rather than the test user).

Is that all that we want to check? What else is included in the composite object?

Reply:
Below is the composite object.  
{
  "permissions": {
    "permissions": [
      "perms.all",
      "okapi.readonly",
      "okapi.all",
      "configuration.all",
      "inventory-storage.all",
      "users.all",
      "user-settings.custom-fields.all",
      "login.all",
      "SYS#mod-login-7.6.0-SNAPSHOT.111#/authn/log/events#[POST]",
      "SYS#mod-login-7.6.0-SNAPSHOT.111#/authn/log/events#[GET]",
      "SYS#mod-login-7.6.0-SNAPSHOT.111#/authn/log/events/{id}#[DELETE]",
      "SYS#mod-login-7.6.0-SNAPSHOT.111#/authn/update#[POST]",
      "SYS#mod-login-7.6.0-SNAPSHOT.111#/authn/password/repeatable#[POST]",
      "SYS#mod-login-7.6.0-SNAPSHOT.111#/authn/login#[POST]",
      "SYS#mod-pubsub-2.5.0-SNAPSHOT.175#/_/tenant#[POST]",
      "event.config.all",
      "SYS#mod-circulation-storage-13.2.0-SNAPSHOT.297#/scheduled-request-expiration#[POST]",
      "SYS#mod-circulation-storage-13.2.0-SNAPSHOT.297#/_/tenant#[POST]",
      "circulation-storage.all",
      "templates.all",
      "SYS#mod-template-engine-1.16.0-SNAPSHOT.75#/templates/{templateId}#[DELETE]",
      "SYS#mod-template-engine-1.16.0-SNAPSHOT.75#/template-request#[POST]",
      "email.message.all",
      "SYS#mod-email-1.13.0-SNAPSHOT.60#/email#[POST]",
      "SYS#mod-sender-1.7.0-SNAPSHOT.31#/message-delivery#[POST]",
      "notify.all",
      "SYS#mod-notify-2.10.0-SNAPSHOT.109#/notify#[POST]",
      "SYS#mod-notify-2.10.0-SNAPSHOT.109#/patron-notice#[POST]",
      "inventory-storage.authorities.all",
      "SYS#mod-feesfines-17.1.0-SNAPSHOT.169#/accounts#[POST]",
      "SYS#mod-feesfines-17.1.0-SNAPSHOT.169#/accounts/{id}#[PUT]",
      "SYS#mod-feesfines-17.1.0-SNAPSHOT.169#/accounts/{id}#[DELETE]",
      "SYS#mod-feesfines-17.1.0-SNAPSHOT.169#/accounts/{id}/pay#[POST]",
      "SYS#mod-feesfines-17.1.0-SNAPSHOT.169#/accounts/{id}/waive#[POST]",
      "SYS#mod-feesfines-17.1.0-SNAPSHOT.169#/accounts/{id}/transfer#[POST]",
      "SYS#mod-feesfines-17.1.0-SNAPSHOT.169#/accounts/{id}/cancel#[POST]",
      "SYS#mod-feesfines-17.1.0-SNAPSHOT.169#/accounts/{id}/refund#[POST]",
      "SYS#mod-feesfines-17.1.0-SNAPSHOT.169#/accounts-bulk/pay#[POST]",
      "SYS#mod-feesfines-17.1.0-SNAPSHOT.169#/accounts-bulk/waive#[POST]",
      "SYS#mod-feesfines-17.1.0-SNAPSHOT.169#/accounts-bulk/transfer#[POST]",
      "SYS#mod-feesfines-17.1.0-SNAPSHOT.169#/accounts-bulk/cancel#[POST]",
      "SYS#mod-feesfines-17.1.0-SNAPSHOT.169#/manualblocks/{id}#[DELETE]",
      "SYS#mod-feesfines-17.1.0-SNAPSHOT.169#/accounts-bulk/refund#[POST]",
      "SYS#mod-feesfines-17.1.0-SNAPSHOT.169#/feefineactions#[POST]",
      "SYS#mod-feesfines-17.1.0-SNAPSHOT.169#/manualblocks#[POST]",
      "SYS#mod-feesfines-17.1.0-SNAPSHOT.169#/manualblocks/{id}#[PUT]",
      "SYS#mod-feesfines-17.1.0-SNAPSHOT.169#/_/tenant#[POST]",
      "SYS#mod-feesfines-17.1.0-SNAPSHOT.169#/feefine-reports/refund#[POST]",
      "SYS#mod-feesfines-17.1.0-SNAPSHOT.169#/feefine-reports/cash-drawer-reconciliation#[POST]",
      "SYS#mod-feesfines-17.1.0-SNAPSHOT.169#/feefine-reports/financial-transactions-detail#[POST]",
      "users.all",
      "perms.users.get",
      "perms.users.item.put",
      "perms.users.assign.immutable",
      "owners.item.post",
      "accounts.item.post",
      "users-bl.item.get",
      "users-bl.transactions.get",
      "users-bl.item.delete"
    ]
  },
  "proxiesFor": [
  ],
  "user": {
    "metadata": {
      "createdDate": "2022-01-17T10:16:23.892+00:00",
      "updatedDate": "2022-01-17T10:16:23.892+00:00"
    },
    "createdDate": "2022-01-17T10:16:23.902+00:00",
    "active": true,
    "personal": {
      "lastName": "Orders API Tests",
      "firstName": "Admin",
      "addresses": [
      ]
    },
    "id": "00000000-1111-5555-9999-999999999991",
    "updatedDate": "2022-01-17T10:16:23.902+00:00",
    "departments": [
    ],
    "proxyFor": [
    ],
    "username": "test-admin"
  }
}

Reply:
Should the test check the permissions or the proxy information?

Reply:
no its not, but i think its needed. i will add those checks . thanks

Reply:
Added permissions check